cdap-ui/app/cdap/components/CopyableID/index.js

```diff
@@ -60,18 +68,18 @@ export default class CopyableRunID extends Component {
     return (
       <span
         className="btn-link"
-        id={runidlabel}
-        onClick={this.onIDClickHandler.bind(this, this.props.runid)}
+        id={idlabel}
+        onClick={this.onIDClickHandler.bind(this, this.props.id)}
         onMouseOut={() => {
           this.state.showTooltip ? this.onIDClickHandler() : null;
         }}
-        data-clipboard-text={this.props.runid}
+        data-clipboard-text={this.props.id}
       >
-        <span>{T.translate(`${PREFIX}.label`)}</span>
+        <span>{this.props.label}</span>
         <UncontrolledTooltip
           {...tooltipProps}
         >
-          <span>{this.props.runid}</span>
+          <span>{this.props.id || T.translate(`${PREFIX}.notAvailable`)}</span>
```

Comment:
So `isRequired` can still render the component. On a dev build it will throw an error in the console saying the id is required but the prop types mismatch didn't show any error when running in production mode.